src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -136,3 +151,39 @@ def _inner_not_found(ex):
 
     with self.command_group('cdn edge-node', cdn_edge_sdk) as g:
         g.command('list', 'list')
+
+    with self.command_group('cdn waf policy', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list')
+        g.custom_command('set', 'set_waf_policy', client_factory=cf_waf_policy)
+        g.command('delete', 'delete')
+
+    with self.command_group('cdn waf policy managed-rule-set', cdn_waf_policy_sdk, is_preview=True) as g:
+        g.custom_command('add', 'add_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('remove', 'remove_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list', 'list_waf_policy_managed_rule_sets', client_factory=cf_waf_policy)
+        g.custom_show_command('show', 'show_waf_policy_managed_rule_set', client_factory=cf_waf_policy)
+        g.custom_command('list-available', 'list_waf_managed_rule_set', client_factory=cf_waf_rule_set)
+        g.custom_command('show-available', 'show_waf_managed_rule_set', client_factory=cf_waf_rule_set)
+
+    with self.command_group('cdn waf policy managed-rule-set rule-group-override',
+                            cdn_waf_policy_sdk,
+                            is_preview=True) as g:
+        g.custom_command('set', 'set_waf_managed_rule_group_override', client_factory=cf_waf_policy)
+        g.custom_command('delete', 'delete_waf_managed_rule_group_override', client_factory=cf_waf_policy)
```

Comment:
it will be better to have `confirmation` for delete command.

Reply:
Fixed